lib/board.rb

```diff
@@ -0,0 +1,60 @@
+require 'pry'
+class Board
+  # Is this saving
+
+  attr_accessor :cells
+
+  def initialize(cells = Array.new(9," "))
+    @cells = cells
+
+  end
+
+  def reset!
+    cells.clear
+    9.times do
+      cells << " "
+    end
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "------------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "------------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(user_input)
+    user_input = user_input.to_i
+    cells[user_input-1]
+  end
+
+  def full?
+    cells.all?{|cell| cell != " "}
+  end
+
+  def turn_count
+    cells.count{|token| token == "X" || token == "O"}
+  end
+
+  def taken?(user_input)
+    user_input = user_input.to_i
+    cells[user_input-1] == "X" || cells[user_input-1] == "O"
+  end
+
+  def valid_move?(user_input)
+    user_input = user_input.to_i
+    !full? && !taken?(user_input) if user_input >= 1 && user_input < 10
```

Comment:
You can simplify checking the valid integer range, and make it more semantic, by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F).
`position.between?(1,9)`

Reply:
Hey I tried this code:
user_input.between?(1,9) 
but I couldn't pass for the input that were taken so I added an if statement
user_input.between?(1,9) if !taken?(user_input)
Is this alright or should I find another way?